data/mods/Xedra_Evolved/mutations/vampire_trait_eocs.json

```diff
@@ -654,6 +654,80 @@
     "condition": { "compare_string": [ "ANATHEMA_WEAKNESS_NEED_VAMPIRE_BLOOD", { "context_val": "trait" } ] },
     "effect": [ { "u_message": "You feel a growing thirst for the blood of vampires.", "type": "bad" } ]
   },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_VAMPIRE_ANATHEMA_BURN_IN_SUNLIGHT",
+    "eoc_type": "EVENT",
+    "required_event": "avatar_moves",
+    "condition": {
+      "and": [
+        { "u_has_trait": "VAMPIRE_ANATHEMA" },
+        { "u_has_trait": "ANATHEMA_WEAKNESS_SUN_BURN" },
+        "is_day",
+        "u_is_outside",
+        { "not": { "is_weather": "blotted_sun" } }
+      ]
+    },
+    "effect": [
+      { "u_add_effect": "effect_slain_by_the_sun", "duration": "PERMANENT" },
+      {
+        "if": { "not": { "u_has_effect": "effect_slain_by_the_sun_limiter" } },
+        "then": [ { "run_eocs": "EOC_VAMPIRE_ANATHEMA_BURN_IN_SUNLIGHT_CONTINUAL" } ]
+      }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_VAMPIRE_ANATHEMA_BURN_IN_SUNLIGHT_GO_INDOORS",
+    "eoc_type": "EVENT",
+    "required_event": "avatar_moves",
+    "condition": {
+      "and": [
+        { "u_has_trait": "VAMPIRE_ANATHEMA" },
+        { "u_has_trait": "ANATHEMA_WEAKNESS_SUN_BURN" },
+        { "or": [ { "not": "is_day" }, { "not": "u_is_outside" }, { "is_weather": "blotted_sun" } ] }
+      ]
+    },
+    "effect": [ { "u_lose_effect": "effect_slain_by_the_sun" }, { "u_lose_effect": "effect_slain_by_the_sun_limiter" } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_VAMPIRE_ANATHEMA_BURN_IN_SUNLIGHT_CONTINUAL",
+    "condition": { "u_has_effect": "effect_slain_by_the_sun" },
+    "effect": [
+      { "u_add_effect": "effect_slain_by_the_sun_limiter", "duration": "PERMANENT" },
+      {
+        "math": [ "_vampire_burning_chance = (30 * (u_effect_intensity('effect_vampiric_resilience') > -1 ? 4 : 1) )" ]
+      },
+      {
+        "foreach": "array",
+        "target": [ "arm_l", "arm_r", "leg_l", "leg_r", "torso", "head" ],
+        "var": { "context_val": "id" },
+        "effect": [
+          {
+            "if": { "x_in_y_chance": { "x": 1, "y": { "math": [ "_vampire_burning_chance" ] } } },
+            "then": [ { "u_add_effect": "onfire", "duration": 30, "target_part": { "context_val": "id" } } ]
+          }
+        ]
+      },
+      {
+        "u_message": "You are burning in the sunlight!  Your burn chance is 1 in <context_val:vampire_burning_chance>.",
+        "type": "debug"
+      },
+      { "run_eocs": "EOC_VAMPIRE_ANATHEMA_BURN_IN_SUNLIGHT_CONTINUAL", "time_in_future": 1 }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_XE_VAMPIRE_ANATHEMA_BURN_IN_SUNLIGHT_AT_SUNRISE",
+    "recurrence": { "math": [ "time_until('sunrise')" ] },
+    "global": true,
+    "run_for_npcs": false,
+    "condition": {
+      "and": [ { "u_has_trait": "VAMPIRE_ANATHEMA" }, "is_day", "u_is_outside", { "not": { "is_weather": "blotted_sun" } } ]
```

Comment:
Checking the weakness instead of the unlock trait would be more straightforward.